### ISSUE TITLE
 ReferenceError: window is not defined

### ISSUE THREAD

**Issue by Owned67** [daniel-sc/bash-shell-to-bat-converter#12]
`
git clone https://github.com/daniel-sc/bash-shell-to-bat-converter.git
Cloning into 'bash-shell-to-bat-converter'...
remote: Enumerating objects: 174, done.
remote: Counting objects: 100% (174/174), done.
remote: Compressing objects: 100% (137/137), done.
remote: Total 174 (delta 70), reused 123 (delta 34), pack-reused 0
Receiving objects: 100% (174/174), 529.01 KiB | 823.00 KiB/s, done.
Resolving deltas: 100% (70/70), done.

cd bash-shell-to-bat-converter

npm i
added 32 packages from 59 contributors and audited 32 packages in 13.766s
found 0 vulnerabilities

npm run tsc

> bash-converter@0.1.0 tsc F:\bash-shell-to-bat-converter
> tsc

node lib/cli.js uasp.sh
..\lib\convert-bash.js:5
window.process = { env: { NODE_NEV: 'mock' } }; // patch for irrelevant node dependency of bash-parser
^

ReferenceError: window is not defined`

Windows 10 x64
nodejs v10.15.1

I've trying online version but it doesn't work....

**Comment by daniel-sc** [comment]
Hi @Owned67 ,

I pushed a fix - would you mind testing with the lastest `master`?

**Comment by Owned67** [comment]
Thanks solved

**Comment by SashiRin** [comment]
It seems this issue was not fixed in my case. I manually add var before window variable in `convert-bash.js` line 7
```
var window = { process ...
```

**Comment by daniel-sc** [comment]
Hi @SashiRin 
Could you provide a complete replication example including OS and nodejs version and the complete error message?

**Comment by SashiRin** [comment]
Yes.
WSL Ubuntu 20.04
Nodejs 14.15.1

```
npm install
npm run tsc
node lib/cli.js FILE_TO_CONVERT.sh
```

**Comment by daniel-sc** [comment]
Hi @SashiRin ,

yes - I could reproduce this. It should be fixed by my last commit - please let me know if this is resolved for you as well!

**Comment by SashiRin** [comment]
It is resolved, great.
Thank you!

**Comment by Arun-Karunakaran** [comment]
Hi I am using this on a mac and I get the below error when I do npm run tsc and unable to proceed further. help me in understanding why I get this error. I have installed npm and typescript (using brew install). Still i get the error Error: Cannot find module 'tslib' and bash-parser. 

$ npm run tsc

> bash-converter@0.1.0 tsc
> tsc

src/convert-bash.ts:10:19 - error TS2307: Cannot find module 'bash-parser' or its corresponding type declarations.

10 import parse from 'bash-parser'
                     ~~~~~~~~~~~~~

src/rm-handler.ts:10:16 - error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.

10         const {singleDashParams, doubleDashParams, argList} = splitParams(command);
                  ~~~~~~~~~~~~~~~~


Found 2 errors.

npm ERR! code 2
